### ISSUE TITLE
download_dsyms does not always find the dsyms when version is set to 'latest'

### ISSUE THREAD

**Issue by splitmo** [fastlane/fastlane#20605]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
download_dsyms does not always find the dsyms when version 'latest' is selected.  
I was able to debug this to the API call to Apple. Actually I believe it's a bug on their end, but I do have a suggestion how to fix it.

<details>

I used the following command on the same account for two different apps, and it works for one yet it doesn't for the other (which is an app that isn't live yet, maybe thats the difference but thats just a guess). 

  <pre>
 download_dsyms(
       app_identifier: "....",
       version: "latest",
       username: "...."
    )
</pre>

Download dsyms first retrieves the latest build info using the iris/v1/builds API call, and then it does another API call using this found build info for the one build to fetch the download URL. The build I am debugging has a prereleaseversion of 4.0.0 and a build number of 463. Although the build is found in the first API call and all metadata set correctly, the second API call does not return this build. This is the call that doesn't retrieve any builds:

<pre>

https://appstoreconnect.apple.com/iris/v1/builds?filter%5Bapp%5D=*****&filter%5BpreReleaseVersion.platform%5D=IOS&filter%5BpreReleaseVersion.version%5D=4.0.0&filter%5Bversion%5D=463

</pre>

I used mitmproxy to see the result from apple, and verified it returned zero results for the second API call. I tried the same API call to see if any results were returned when I removed any of the filters, and it actually returned the result when I removed the prereleaseversion.version filter:

<pre>
https://appstoreconnect.apple.com/iris/v1/builds?filter%5Bapp%5D=*****&filter%5BpreReleaseVersion.platform%5D=IOS&filter%5Bversion%5D=463
</pre>

Now any sane developer would think, the prereleaseversion specified must be wrong. I triple checked this, but I don't see anything wrong with it. The returned value of the first API call returns 4.0.0. It is 4.0.0... I'm clueless why this works for one app but doesn't for the other.

How to fix? As I believe a build can be uniquely identified by app id and buildnumber, I think the prereleaseversion is not needed if both app id and build number are present. Hence I would suggest to apply the preReleaseVersion.version filter only when the build number is not present:

<pre>
        - filter["preReleaseVersion.version"] = version if version
        + filter["preReleaseVersion.version"] = version if version && !build_number
 </pre>

I patched and tested download_dsyms.rb which worked:

<pre>
[21:09:28]: 🔑  Successfully downloaded dSYM file for 4.0.0 - 463 to 'XXXXXXX-4.0.0-463.dSYM.zip'
</pre>

</details>

**Comment by adarhef** [comment]
From what I understand, there are no more downloadable dsyms for builds submitted with Xcode 14, probably related to bitcode being deprecated.
Here's a thread on the official dev forums: [https://developer.apple.com/forums/thread/716826](url)
Apple rep claims dsyms no longer need to be downloaded from Apple, the ones included in the local archive can be used for uploading to Crashlytics immediately after build.

**Comment by mjooms** [comment]
Yes that is correct; if you don't use bitcode you can't (and won't need) to download the DSYMS from Apple anymore because they don't recompile the binary, and thus your own DSYMS can be used directly. 

Since probably no one will be using bitcode in the near future I guess this ticket can be closed.
